### PR TITLE
Project View Search Button

### PR DESCRIPTION
**Pull request by thisisnian**
This is to create a 'Find' button so that users are able to press Find instead of having to press enter.
This is to improve user friendiness, instead of having to guess enter = find.

This is my first dabble in coffee, code review appreciated. 

Please see https://github.com/atom/find-and-replace/issues/297

Cheers,

----

Added by @simurai

Before:

screen shot 2015-09-22 at 10 48 29 pm

After:

screen shot 2015-09-22 at 10 46 07 pm

Once this gets merged, also merge One dark/light PRs https://github.com/atom/one-dark-ui/issues/91 + https://github.com/atom/one-light-ui/issues/35.

### CODE REVIEW

**Comment by mertkahyaoglu** [comment]
IMO, just pressing enter is fast and very user friendly :grin: but let's see anyone want this. The only thing bothers me about the code is additional line breaks.

**Comment by thisisnian** [comment]
Well there have been some users (myself included) that wasn't sure how to search, either this or even an 'Enter to Search' tooltip or something. 

In any case, users can still press the enter button, whatever floats your boat.

I've just fixed the formatting.

File: lib/project-find-view.coffee [hunk 1]
```diff
@@ -365,4 +374,4 @@ class ProjectFindView extends View
     @search(onlyRunIfActive: true, caseSensitive: not @model.getFindOptions().caseSensitive)
 
   toggleWholeWordOption: ->
-    @search(onlyRunIfActive: true, wholeWord: not @model.getFindOptions().wholeWord)
+    @search(onlyRunIfActive: true, wholeWord: not @model.getFindOptions().wholeWord)
```
Comment (50Wliu) [inline lib/project-find-view.coffee]: This space isn't needed and also causes the linter tests to fail.

**Comment by thisisnian** [comment]
thanks must have missed that one.

**Comment by simurai** [comment]
How about changing the button widths a bit so that the inputs line up? Something like:

screen shot 2015-08-26 at 4 50 19 pm

I can help out with the styling if you like it.

**Comment by thisisnian** [comment]
@simurai Got it like the screenshot above, and I've got it built in Linux. However it fails Travis CI. I guess I'm doing something wrong?

**Comment by simurai** [comment]
I think the check fails because `project-find` got renamed to `find-and-replace`.

**Comment by thisisnian** [comment]
I reverted the failing changes on my local instance, this is what I got 

image

I'm not pretty sure where the property (boxed in red) comes from. Looks like it may be from another package as the one defined in the styles gets overwritten. Has to be more than 10em otherwise the buttons are too small. 

Gah I feel so stupid, must be something really simple that I'm missing.

**Comment by simurai** [comment]
> I'm not pretty sure where the property (boxed in red) comes from. Looks like it may be from another package as the one defined in the styles gets overwritten.

Yes, right. That comes from the One dark theme. If you switch to another theme (like Atom dark), then it should look fine. I'll make another PR for the One dark theme to fix it. It needs the `em`s so that font sizing can be changed.

__Tip__: In case you'd like to see where the styles come from, you can click the `<style>...</style>`, then it should highlight the less file.

screen shot 2015-08-29 at 12 20 12 pm
screen shot 2015-08-29 at 12 20 44 pm

**Comment by thisisnian** [comment]
Ah I knew it came from there, however I was just unsure as it would mean potentially not just one-dark theme that needs changing, however anyone with a custom theme that didn't have the correct css attributes (could be many in the future)

Thanks for checking over it.

**Comment by simurai** [comment]
> I was just unsure as it would mean potentially not just one-dark theme that needs changing, however anyone with a custom theme that didn't have the correct css attributes (could be many in the future)

Typically, themes don't have to bother about styling those buttons. Unless they wanna customize it. The One themes override some of the `px` values to make it possible to change the font size so the whole UI can be resized. But that shouldn't be the concern of this package.

It *is* a bit of a problem when people fork the One themes, since those themes would need a fix too.

**Comment by simurai** [comment]
@benogle How do you like this change of adding a `Find` button to project search? I quite like it:

- Better discovery for beginners
- More consistent with the find in buffer.

**Comment by benogle** [comment]
I think it makes sense. It would be nice if there were a test. Other than than, the code looks fine.

**Pull request merged**